pkg/cluster/rook_ceph.go

```diff
@@ -41,31 +41,44 @@ func init() {
 }
 
 // returns the number of nodes used for storage, which may be higher than the number of names passed in
-// if a node is currenlty not ready but has not been purged
+// if a node is currently not ready but has not been purged
 func (c *Controller) UseNodesForStorage(rookVersion semver.Version, names []string) (int, error) {
 	cluster, err := c.GetCephCluster(context.TODO())
 	if err != nil {
 		return 0, errors.Wrapf(err, "get CephCluster config")
 	}
+	var next []cephv1.Node
 	storageNodes := make(map[string]bool, len(cluster.Spec.Storage.Nodes))
 	for _, storageNode := range cluster.Spec.Storage.Nodes {
+		next = append(next, storageNode)
 		storageNodes[storageNode.Name] = true
 	}
 	changed := false
 	for _, name := range names {
 		if !storageNodes[name] {
 			c.Log.Infof("Adding node %q to CephCluster node storage list", name)
-			cluster.Spec.Storage.Nodes = append(cluster.Spec.Storage.Nodes, cephv1.Node{
+			next = append(next, cephv1.Node{
 				Name: name,
 			})
 			changed = true
 		}
 	}
 	if changed {
-		cluster.Spec.Storage.UseAllNodes = false
-		_, err := c.Config.CephV1.CephClusters("rook-ceph").Update(context.TODO(), cluster, metav1.UpdateOptions{})
+		patches := []k8s.JSONPatchOperation{}
+		patches = append(patches, k8s.JSONPatchOperation{
```

Comment:
Why are we doing the patches via creating the endpoints instead of use [client.Patch](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/typed_client.go#L114) ?